pkg/client/metrics/metrics.go

```diff
@@ -44,15 +44,35 @@ var (
 			Buckets:   prometheus.DefBuckets,
 		},
 		[]string{"path", "server"},
+
+	)
+
+	// define a counter for API errors for various ErrorTypes
+	apiErrorCount = metrics.NewCounterVec(
+    		&metrics.CounterOpts{
+        		Namespace: "prometheus_adapter",
+        		Subsystem: "prometheus_client",
+        		Name:      "api_errors_total",
+        		Help:      "Total number of API errors",
+    		},
+    		[]string{"error_code", "path", "server"},
+
 	)
 )
 
 func MetricsHandler() (http.HandlerFunc, error) {
 	registry := metrics.NewKubeRegistry()
-	err := registry.Register(queryLatency)
-	if err != nil {
-		return nil, err
+
+	errRegisterQueryLatency := registry.Register(queryLatency)
+	if errRegisterQueryLatency != nil {
+		return nil, errRegisterQueryLatency
+	}
+
+	errRegisterAPIErrorCount := registry.Register(apiErrorCount)
+	if errRegisterAPIErrorCount != nil {
+		return nil, errRegisterAPIErrorCount
 	}
```

Comment:
Since we don't necessarily need to differentiate between the errors here, it is usually a common practice in Go to just have one `err` variable and reuse it for reach `Register` call.

Reply:
updated, thanks